website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,62 @@
-# Widget Errors
+# Widget errors
 
-This section will help you troubleshoot common widget errors on the Appsmith platform.
+This section would help you troubleshoot common widget errors on the Appsmith platform.
 
-## JSON Form Errors
+## JSON form errors
 
 You may see the below errors when working with a [JSON Form ](../../reference/widgets/json-form)widget.
 
-### **Source data exceeds 50 fields**
+### Source data exceeds 50 fields
 
 You see an error message **Source data exceeds 50 fields. Please update the source data** when you try to bind the query/API response to the [source data ](../../reference/widgets/json-form#source-data)property.
 
-**Error message:**
+#### Error message
 
 Source data exceeds 50 fields. Please update the source data.
 
 ![When the data had more than 50 fields](</img/Troubleshooting__Widget_Errors__JSON_Form_Errors__Source_Exceeds_50_Fields.png>)
 
-### Cause
+#### Cause
 
 The problem can be caused when you are trying to bind:
 
 * A large array of multiple JSON objects
 * A huge JSON object which has a lot of fields
 * The whole query data rather than a selected row or triggered row in a table
 
-### Solution
+#### Solution
 
 To determine if the problem is caused due to:
 
-* **A large array or a huge JSON object** - You can relook at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
-* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either selected row / triggered row to bind.
+* **A large array or a huge JSON object** - You can re-look at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
+* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either the selected row / triggered row to bind.
 
 Once you have figured out the new structure for the data, head to the [source data](../../reference/widgets/json-form#source-data) field to make changes.
 
+## Query/API/JS object execution errors
+You may see below errors when executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Action can't be triggered
+You may see this error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
+
+#### Error message
+Action cannot be triggered from this field.
```

Comment:
Error messages have been reiterated after a discussion with the team. 

Please check here for more information https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1667457021297869?thread_ts=1667385039.225229&cid=C02K0SZQ7V3

Reply:
@Rishabh-Rathod - This is what I gathered from the conversation. The error message is updated as below:

`Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.`

Please confirm so that I can update the PR.

CC: @bharath31

Reply:
- Below is the error message when an action like storeValue is used in the sync field.

```Found a reference to {{action}} during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to {{action}} and try again.```

- When the user writes a JS code in sync fields that returns a promise 

For such cases, as sync fields cannot execute asynchronous code we show the below error message. 

```Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.```

Reply:
Below is the error message when an action like storeValue is used in the sync field.
Error message - 1 --> `Found a reference to {{action}} during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to {{action}} and try again.`

Will the `{{action}}` in this example will be replaced with `storeValue`?

For such cases, as sync fields cannot execute asynchronous code we show the below error message.
ErrorMessage - 2 --> `Found a Promise() during evaluation. Sync fields cannot execute asynchronous code.`

A bit confused with the above error messages. For users on the UI which error message will be shown Error message 1 or error message 2.

Reply:
1. `Will the {{action}} in this example will be replaced with storeValue?`

Yes, to be precise it will be replace with `storeValue()`.  

Example:-
```Found a reference to storeValue() during evaluation. Sync fields cannot execute async framework actions. Please remove any direct/indirect references to storeValue()  and try again.```

Reply:
Thanks @Rishabh-Rathod. And when will the error message 2 be displayed to users.

Reply:
When a user writes code something like ```{{JSObject.myFun1()}}``` in the sync field where `myFun1` is a javascript async function ( function defined with `async` keyword )

Reply:
@Rishabh-Rathod - Do you think we should add these two error messages as two separate troubleshooting errors? CC: @danciaclara - your thoughts??

Reply:
Yes, we should add both errors to help user understand.

Reply:
Updated error messages.